tokio/src/time/driver/entry.rs

```diff
@@ -420,20 +421,21 @@ impl TimerShared {
 /// padded. This contains the information that the driver thread accesses most
 /// frequently to minimize contention. In particular, we move it away from the
 /// waker, as the waker is updated on every poll.
+#[repr(C)] // required by `link_list::Link` impl
 struct TimerSharedPadded {
+    /// A link within the doubly-linked list of timers on a particular level and
+    /// slot. Valid only if state is equal to Registered.
+    ///
+    /// Only accessed under the entry lock.
+    pointers: linked_list::Pointers<TimerShared>,
```

Comment:
am i correct that pointers no longer need to be in an `UnsafeCell`, since they're only accessed via a raw pointer cast that assumes they're the struct's first field? it might be worth adding to the comment that they can only be accessed when holding the lock *and* only via pointer casts?

Reply:
Well, the reason it no longer needs to be wrapped in an `UnsafeCell`, is that the fields inside it are wrapped in an `UnsafeCell`.

Reply:
oh, did that change in a different branch? i must have missed it. thanks for clearing that up!

Reply:
That happened in #3654.